src/internal/connector/onedrive/collection.go

```diff
@@ -65,7 +65,10 @@ type Collection struct {
 	folderPath path.Path
 	// M365 IDs of file items within this collection
 	driveItems map[string]models.DriveItemable
-	// M365 ID of the drive this collection was created from
+	// Map of M365 Drive/Names associated with collection
+	// key M365ID: Value DriveName
+	driveMap map[string]string
```

Comment:
A collection always belongs to the same Drive - shouldn't this just be a `driveName` string. 

Also - should we just query and persist this in the `Collections` struct? Collections is also 1:1 with a Drive

Reply:
We should just query this metadata when we call `drives`

Reply:
https://github.com/alcionai/corso/blob/main/src/internal/connector/onedrive/collections.go#L275

Reply:
The collection does not always belong to the same Drive. `sharePoint` has several different drives associated.

Reply:
I understand - but does each collection belong to multiple drives?

Reply:
Personal OneDrives are 1:1
SharePoint  OneDrives are 1 collection to Many
In some cases, it is one. But sometimes it can be more than that.

Reply:
With the changes applied, there is one additional call per existing drive for SharePoint. 

For OneDrive, there are no additional calls.
I tested by running this with the `Corso` binary and viewing the logs that were added each time the call is requested.

Reply:
> For a site, a developer can choose to create an additional document library which will result in an additional drive. For OneDrive, a user can create an additional document library; however, it is not discovered during backup. These newly created drives are part of the shared space which we do not have coverage for as of yet.

I understand this - but a collection (which is a folder in a drive) - only belongs to a single drive - correct?

Reply:
Commit [db8bd41](https://github.com/alcionai/corso/pull/2733/commits/db8bd41783d48e9fe7912911344b5241822e799e) creates an efficient pathway for `OneDrive` or any other features where we know there is only one available document library.

Reply:
There is a bug within the `UpdateCollections`. Working to resolve

Reply:
Issue #2740 filed